FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,418 @@
+  fi
+}
+
+# Configure the use of FreeSurfer v6 custom tools
+configure_custom_tools
+
+# Show tool versions
+show_tool_versions
+
+# Validate version of FreeSurfer in use
+validate_freesurfer_version
+
+# ----------------------------------------------------------------------
+log_Msg "Starting main functionality"
+# ----------------------------------------------------------------------
+
+
+# ----------------------------------------------------------------------
+# Log values retrieved from positional parameters
+# ----------------------------------------------------------------------
+log_Msg "StudyFolder: ${StudyFolder}"
+log_Msg "SubjectID: ${SubjectID}"
+log_Msg "Sessions: ${Sessions}"
+log_Msg "TemplateID: ${TemplateID}"
+log_Msg "recon_all_seed: ${recon_all_seed}"
+log_Msg "extra_reconall_args_base: ${extra_reconall_args_base[*]+"${extra_reconall_args_base[*]}"}"
+log_Msg "extra_reconall_args_long: ${extra_reconall_args_long[*]+"${extra_reconall_args_long[*]}"}"
+
+# ----------------------------------------------------------------------
+log_Msg "Preparing the folder structure"
+# ----------------------------------------------------------------------
+Sessions=`echo ${Sessions} | sed 's/@/ /g'`
+log_Msg "After delimiter substitution, Sessions: ${Sessions}"
+LongDIR="${StudyFolder}/${SubjectID}.long.${TemplateID}/T1w"
+mkdir -p "${LongDIR}"
+for Session in ${Sessions} ; do
+  Source="${StudyFolder}/${Session}/T1w/${Session}"
+  Target="${LongDIR}/${Session}"
+  log_Msg "Creating a link: ${Source} => ${Target}"
+  ln -sf ${Source} ${Target}
+done
+
+
+if (( generate_timepoints_only != 1 )); then 
+
+	# ----------------------------------------------------------------------
+	log_Msg "Creating the base template: ${TemplateID}"
+	# ----------------------------------------------------------------------
+	# backup template dir if it exists
+	if [ -d "${LongDIR}/${TemplateID}" ]; then
+	  TimeStamp=`date +%Y-%m-%d_%H.%M.%S.%6N`
+	  log_Msg "Base template dir: ${LongDIR}/${TemplateID} already exists, backing up to ${LongDIR}/${TemplateID}.${TimeStamp}"
+	  mv ${LongDIR}/${TemplateID} ${LongDIR}/${TemplateID}.${TimeStamp}
+	fi
+
+	recon_all_cmd="recon-all.v6.hires"
+	recon_all_cmd+=" -sd ${LongDIR}"
+	recon_all_cmd+=" -base ${TemplateID}"
+	for Session in ${Sessions} ; do
+	  recon_all_cmd+=" -tp ${Session}"
+	done
+	recon_all_cmd+=" -all"
+
+	if [ ! -z "${recon_all_seed}" ]; then
+	  recon_all_cmd+=" -norandomness -rng-seed ${recon_all_seed}"
+	fi
+
+
+	#---------------------------------------------------------------------------------------
+	log_Msg "Running the recon-all to generate common template"
+	#---------------------------------------------------------------------------------------
+
+	#recon_all_cmd+=(${extra_reconall_args_base[@]+"${extra_reconall_args_base[@]}"})
+	recon_all_cmd+=" ${extra_reconall_args_base[@]}"
```

Comment:
I don't see anything related to the use of the T2/FLAIR.  Is that because they are automatically used in the long pipeline if they were used in the cross?

Reply:
I guess this is the template generation section.  So the T2/FLAIR play no role in the generation of the template?